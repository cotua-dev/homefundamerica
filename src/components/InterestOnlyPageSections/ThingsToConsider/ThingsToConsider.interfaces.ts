export interface FetchThingsToConsiderImageQuery {
    thingsToConsiderImage: {
        publicURL: string;
    };
}
