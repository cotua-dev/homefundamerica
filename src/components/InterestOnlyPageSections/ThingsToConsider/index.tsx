import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { FetchThingsToConsiderImageQuery } from './ThingsToConsider.interfaces';
import { thingsToConsiderSection } from './ThingsToConsider.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const ThingsToConsider: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchThingsToConsiderImageQuery>(graphql`
        query {
            thingsToConsiderImage: file(name: {eq: "AdobeStock_201491395"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section
            id={thingsToConsiderSection}
            className={sharedStyles.loanNegative}
        >
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Things to Consider with Interest-Only Mortgages`}</h2>
                <p>
                    {`
                        Interest-Only Mortgages can provide some unique benefits,
                        but they are not right for everyone.
                    `}
                </p>
                <p>
                    {`
                        Things to be conscious of when considering an interest-only
                        option:
                    `}
                </p>
                <ul>
                    <li>{`The interest-only payment does not pay down the principal balance`}</li>
                    <li>{`The monthly payment following the interest-only period will be higher than that of a fully amortized loan that was originated at the same time`}</li>
                    <li>{`The monthly payment will increase after the initial interest-only term ends`}</li>
                </ul>
            </div>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.thingsToConsiderImage.publicURL}
                    alt="Things to consider image"
                />
            </div>
        </section>
    );
}
