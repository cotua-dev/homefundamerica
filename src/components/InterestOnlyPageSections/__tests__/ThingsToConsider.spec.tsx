import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { ThingsToConsider } from '../ThingsToConsider';

describe('ThingsToConsider Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<ThingsToConsider/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
