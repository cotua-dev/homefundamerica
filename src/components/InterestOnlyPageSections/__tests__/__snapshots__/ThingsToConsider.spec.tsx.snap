// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThingsToConsider Component renders 1`] = `
<section
  id="thingsToConsiderSection"
>
  <div>
    <h2>
      Things to Consider with Interest-Only Mortgages
    </h2>
    <p>
      
                        Interest-Only Mortgages can provide some unique benefits,
                        but they are not right for everyone.
                    
    </p>
    <p>
      
                        Things to be conscious of when considering an interest-only
                        option:
                    
    </p>
    <ul>
      <li>
        The interest-only payment does not pay down the principal balance
      </li>
      <li>
        The monthly payment following the interest-only period will be higher than that of a fully amortized loan that was originated at the same time
      </li>
      <li>
        The monthly payment will increase after the initial interest-only term ends
      </li>
    </ul>
  </div>
  <div>
    <img
      alt="Things to consider image"
    />
  </div>
</section>
`;
